src/Runner/TestSuiteSorter.php

```diff
@@ -79,6 +141,38 @@ private function randomize(array $tests): array
         return $tests;
     }
 
+    private function sortDefectsFirst(array $tests): array
+    {
+        \usort($tests, [self::class, 'cmpDefectPriorityAndTime']);
+
+        return $tests;
+    }
+
+    /**
+     * Comparator callback function to sort tests for "reach failure as fast as possible":
+     * 1. sort tests by defect weight defined in self::DEFECT_SORT_WEIGHT
+     * 2. when tests are equally defective, sort the fastest to the front
+     * 3. do not reorder successful tests
+     */
+    private function cmpDefectPriorityAndTime(Test $a, Test $b): int
+    {
+        $priorityA = $this->defectSortOrder[$a->getName()] ?? 0;
```

Comment:
this method is called like that:
`\usort($tests, [self::class, 'cmpDefectPriorityAndTime']);`

meaning, it's called on class, not on instance - like it would be static method. yet, it's not declared as static, and it's accessing `$this->...` inside.
please decide, shall it be static or not, and declare/call it properly

Reply:
It is meant to be a private utility function used to work on a specific instance / runtime context. It requires knowledge of the state the cache and sort order to do its work.

Reply:
I have fixed the static call by the `usort()`